cmd/operator-sdk/alpha/bundle/build.go

```diff
@@ -0,0 +1,103 @@
+package bundle
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/operator-framework/operator-registry/pkg/lib/bundle"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+// newBundleBuildCmd returns a command that will build operator bundle image.
+func newBundleBuildCmd() *cobra.Command {
+	c := bundleCmd{}
+	cmd := &cobra.Command{
+		Use:   "build",
+		Short: "Build an operator bundle image",
+		Long: `The 'operator-sdk bundle build' command will build an operator
+bundle image containing operator metadata and manifests, tagged with the
+provided image tag. Metadata generated by this command defines this image
+and should not be modified.
+
+To write metadata and a bundle image Dockerfile to disk, set '--generate-only[=true]'
+Bundle metadata will be generated in <directory-arg>/metadata, and the Dockerfile
+in <directory-arg>. This flag is useful if you want to build the operator's
+bundle image manually.
+
+NOTE: bundle images are not runnable.
+`,
+		Example: `The following command will build a test-operator bundle image using Docker.
+This image will contain manifests for package channels 'stable' and 'beta':
+
+$ operator-sdk bundle build \
+    --directory ./deploy/olm-catalog/test-operator \
+    --tag quay.io/example/operator:v0.1.0 \
+    --package test-operator \
+    --channels stable,beta \
+    --default stable \
+    --overwrite
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			channels := strings.Join(c.channels, ",")
+			if c.generateOnly {
+				return bundle.GenerateFunc(c.directory, c.packageName, channels,
+					c.channelDefault, true)
+			}
+			// An image tag is required for build only.
```

Comment:
wdyt about the logic to build the image be encapsulated in a func with this responsibility?

Reply:
The func as-is has low enough complexity/LOC that refactoring it into its own func is premature IMO.